Acceptor Dynamic Session Templates

pvkv1799 opened the pull request:
Enables "*" wildcard in Acceptor session configuration (for BeginString, SenderCompID, SenderSubID, SenderLocationID, TargetCompID, TargetSubID, and TargetLocationID).
Such session templates can be defined either at start-up or as an ad-hoc operation, just like "normal" sessions.
When processing a new incoming connection, if incoming SessionID is not equal to any of the created sessions, it is then matched considering wildcards. If such a match is found, the new session is created with the same settings as the template one, just changing wildcards to the actual values from incoming SessionID.

LittleAlexey committed:
Logout can be received when not logged on.

https://github.com/connamara/quickfixn/issues/466

pvkv1799 committed:
Added "AcceptorTemplate" parameter

pvkv1799 committed:
Comment

pvkv1799 committed:
Added SessionID wildcards handling to FileLog

pvkv1799 committed:
Added FileLog unit tests for wildcards

pvkv1799 committed:
Refactored ThreadedSocketAcceptor unit tests: changed to pass configuration string as a parameter instead of the hard-coded constant

pvkv1799 committed:
Added SessionID wildcards handling to FileStore;
Added wildcards validation into SessionSettings;
Moved WILDCARD_VALUE from SessionSettings to Values;
Added Unit Tests for wildcards for the recently changed classes

pvkv1799 committed:
Bug fix: validation of BeginString regardless of Connection Type

pvkv1799 committed:
Bug fix in test: wildcards are valid only for acceptor

pvkv1799 committed:
Added logic to create the new session from a template upon receiving incoming connection

pvkv1799 committed:
Added unit tests for dynamic session templates

pvkv1799 committed:
Removed unused AcceptorTemplate constant

pvkv1799 committed:
Documentation comments for GetSession

gbirchmeier reviewed (changes requested):
Ok, first off, this is going to have some conflicts with #609.  That's not a big PR, but it does make a few changes to Session and SessionFactory.  I hope to merge #609 today.

I'm not crazy about the acceptor-specific additions to Session.  Can we think of any other approach?    Making a DynamicSession subclass seems possible off the top of my head, but also could be horrible.

Don't go deep into any big changes yet.  I also want to talk to @mgatny and see what ideas he has.

pvkv1799 committed:
Removed TODO from GetSession

pvkv1799 committed:
Added check for null _acceptor in CreateFromTemplate

pvkv1799 committed:
Improved explanations in comments

gbirchmeier commented:
FYI, #609 is in master now, so at your leisure you can upmerge or rebase.

pavka1799 commented:
I understood that I should wait for decision on architecture - is the
approach with passing acceptor into a session good enough? Or should it be
changed, probably by adding a class for "template" session?

On Tue, 16 Jun 2020, 17:54 Grant Birchmeier, <notifications@github.com>
wrote:

> FYI, #609 <https://github.com/connamara/quickfixn/pull/609> is in master
> now, so at your leisure you can upmerge or rebase.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/connamara/quickfixn/pull/607#issuecomment-644852954>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKABHZT5FA7TSPBMOK4MIE3RW6IU5ANCNFSM4NR74BKA>
> .
>

gbirchmeier commented:
Haven't had time to think about reworking the impl yet, but for now maybe just resolve the conflicts.

pavka1799 commented:
OK, will resolve

pvkv1799 committed:
Merge pull request #1 from connamara/master

Catching up with the original master

pvkv1799 committed:
Using the new Session constructor with isInitiator

pvkv1799 committed:
Removed commented ACCEPTOR_TEMPLATE constant

pvkv1799 committed:
Renamed GetSession to LookupOrCreateDynamicSession

pvkv1799 committed:
Removed commented-out irrelevant code

pavka1799 commented:
Conflicts resolved, and some of the suggested fixes made (marked them resolved). The rest will do after decision on the architecture.
Thank you!

pvkv1799 committed:
Redirected stderr to stdout (nul) to avoid confusing message "'pwsh' is not recognized as an internal or external command, operable program or batch file."

gbirchmeier committed:
comment about where pwsh comes from

gbirchmeier committed:
fix some typos

gbirchmeier committed:
don't use "./", don't auto-upload nugets

gbirchmeier committed:
restore RELEASE-README.md... again

gbirchmeier committed:
fix formatting of RELEASE_README

dependabot-preview[bot] committed:
Bump NUnit3TestAdapter from 3.16.1 to 3.17.0

Bumps [NUnit3TestAdapter](https://github.com/nunit/nunit3-vs-adapter) from 3.16.1 to 3.17.0.
- [Release notes](https://github.com/nunit/nunit3-vs-adapter/releases)
- [Commits](https://github.com/nunit/nunit3-vs-adapter/compare/V3.16.1...V3.17)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

gbirchmeier committed:
Don't reflect on non-QF/n assemblies

 #630

gbirchmeier committed:
incremented version on release notes

forgot to do this on 1.10.0 release, oops

gbirchmeier committed:
forgot to own it

pavka1799 commented:
@gbirchmeier  Apologies - have you had a chance to think about the architecture? I could do the changes if you have a better solution in mind.
Thank you!

gbirchmeier committed:
corrections for ruby 2.5+; some script annotations

gbirchmeier committed:
remove old unused C++ build

gbirchmeier committed:
use ruby 2.7 on appveyor

gbirchmeier committed:
some debugging for ruby on appveyor

gbirchmeier committed:
more debugging

gbirchmeier committed:
change to ruby 2.6, since appveyor's being weird

2.7 is only on the VS2019 env, but appveyor
is not letting me change to that env.

gbirchmeier committed:
ignore .vscode/

gbirchmeier committed:
add 1156 (ApplExtId) to FIXT11.xml

 #643

gbirchmeier committed:
stupid copy/paste error in previous commit

 #643

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/connamara/quickfixn?pullRequest=607) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/connamara/quickfixn?pullRequest=607) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: LittleAlexey<br/>:white_check_mark: gbirchmeier<br/>:white_check_mark: pvkv1799<br/>:x: dependabot-preview[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/connamara/quickfixn?pullRequest=607) it.</sub>

gbirchmeier commented:
replaced by #650
